File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {
        if(head == null) return null;
        ArrayDeque<ListNode> st = new ArrayDeque<>();
        ListNode pointer = head;
        while (pointer!=null) {
            st.push(pointer);
            pointer = pointer.next;
        }
        head = st.pop();
        pointer = head;
        while(!st.isEmpty()){
            pointer.next = st.peek();

            pointer = st.pop();
        }
        pointer.next = null;
        return head;
    }
}

File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode dummy;
        ListNode* pointer = &dummy;
        while(list1!=NULL && list2!=NULL){
            if(list1->val>list2->val){
                pointer->next = list2;
                list2 = list2->next;
            }else{
                pointer->next = list1;
                list1 = list1->next;
            }
            pointer = pointer->next;
        }
        if(list1==NULL) pointer->next = list2;
        else if(list2==NULL) pointer->next = list1;

        return dummy.next;
    }
};
